src/bin/cargo/commands/tree.rs

```diff
@@ -152,6 +157,16 @@ pub fn exec(config: &mut Config, args: &ArgMatches<'_>) -> CliResult {
     let (edge_kinds, no_proc_macro) = parse_edge_kinds(config, args)?;
     let graph_features = edge_kinds.contains(&EdgeKind::Feature);
 
+    let pkgs_to_prune = args.values_of("prune").map_or_else(
+        || Vec::new(),
+        |ps| {
+            let mut specs = ps.map(|p| p.to_string()).collect::<Vec<_>>();
+            specs.sort_unstable();
+            specs.dedup();
+            specs
+        },
+    );
```

Comment:
Is there any particular reason to deduplicate this?  I would think it should be fine if there are duplicates, and I don't think the rest of cargo does anything similar.  An alternative, just to keep the code smaller, would be:

```suggestion
    let pkgs_to_prune = args._values_of("prune");
```